src/USER-CONP/ewald_conp.cpp

```diff
@@ -1914,19 +1915,64 @@ void EwaldConp::compute_vector_corr(bigint *imat, double *vec) {
     else
       eik_dot_r_triclinic();
   }
-  // TODO 2D correction
+  if (wireflag) {
+    error->all(FLERR, "wire correction not implemented for b vector");
+  }
   int const nlocal = atom->nlocal;
+  int const nprocs = comm->nprocs;
   double **x = atom->x;
   double *q = atom->q;
-  double dipole = 0.;
-  for (int i = 0; i < nlocal; i++) {
-    if (imat[i] < 0) dipole += q[i] * x[i][2];
-  }
-  MPI_Allreduce(MPI_IN_PLACE, &dipole, 1, MPI_DOUBLE, MPI_SUM, world);
-  dipole *= 4.0 * MY_PI / volume;  // TODO why 4? should be 2
-  for (int i = 0; i < nlocal; i++) {
-    int const pos = imat[i];
-    if (pos >= 0) vec[pos] += x[i][2] * dipole;
+  if (slabflag == 1) {
+    // use EW3DC slab correction
+    double dipole = 0.;
+    for (int i = 0; i < nlocal; i++) {
+      if (imat[i] < 0) dipole += q[i] * x[i][2];
+    }
+    MPI_Allreduce(MPI_IN_PLACE, &dipole, 1, MPI_DOUBLE, MPI_SUM, world);
+    dipole *= 4.0 * MY_PI / volume;  // TODO why 4? should be 2
```

Comment:
Uh, weiß ich auch nicht genau, warum das 4 sein muss. Guck dir vielleicht nochmal die slabcorr() in ewald.cpp an?

Reply:
Da hatten wir schonmal drueber geredet. Ich nehm das raus